### PR TITLE
Cadams/interpreter fixes

### PR DESCRIPTION
**Pull request by chris-chen-creates**
Opening a PR for the interpreter and coming tests to be added

### CODE REVIEW

File: src/__tests__/Interpreter.test.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  Negative,
+  Group,
+  Num,
+  Division,
+} from '../Expression'
+import { Interpreter } from '../Interpreter'
+import { parse } from '../Parser'
+import { scan } from '../Scanner'
+
+test('addition works correctly', () => {
```
Comment (angles-n-daemons) [inline src/__tests__/Interpreter.test.ts:13]: Can you add some tests for grouping? And longer expressions like:

`3 + 4 + 5 * 8`

File: src/__tests__/Parser.test.ts [hunk 2]
```diff
@@ -155,13 +155,15 @@ test('parse throws an error if there are unparsed tokens', () => {
...
-  let parser = new Parser(scan('5 + 4'))
-  expect(() => {
-    parser.multiplication()
-  }).toThrow(/.*Addition is not valid multiplication.*/)
-})
-
-test('is multiplication also valid addition?', () => {
-  let parser = new Parser(scan('5 * 4'))
-})
+// Verify that the parser is not at the end
```
Comment (angles-n-daemons) [inline src/__tests__/Parser.test.ts]: why remove these tests?

**Review by angles-n-daemons — CHANGES_REQUESTED** [review]
Couple of comments

File: src/__tests__/Parser.test.ts (hunk 2 above)
Comment (chris-chen-creates) [reply src/__tests__/Parser.test.ts]: These are still in-progress 😅

File: src/__tests__/Interpreter.test.ts (hunk 1 above)
Comment (chris-chen-creates) [reply src/__tests__/Interpreter.test.ts:13]: Added a bunch more tests of varying complexity

File: src/__tests__/Parser.test.ts (hunk 2 above)
Comment (chris-chen-creates) [reply src/__tests__/Parser.test.ts]: Got that fixed and checked to see if the parser was reaching the end of the input

**Review by angles-n-daemons — APPROVED** [review]
Looks good :)

**Pull request merged**